presets.js

```diff
@@ -1,215 +1,370 @@
+      `Capture${i}`,
+      combineRgb(200, 180, 0)
+    );
+  }
 
-	// Still Images - Display
-	for (var i = 1; i <= 10; i++) {
-		presets.push(getPresetForStillImageDisplay(`Display Image ${i}`, `Disply ${i}`, `Display${i}`, self.rgb(0, 90, 0), self.rgb(255, 0, 0)));
-	}
-	presets.push(getPresetForStillImageDisplay('Display Test Image', 'Test image', 'DisplayTest', self.rgb(153, 0, 153), self.rgb(255, 0, 0), '18'));
-	presets.push(getPresetForStillImageDisplay('Display Blackout', 'Blackout', 'Blackout', self.rgb(0, 0, 0), self.rgb(255, 0, 0), '14'));
-	presets.push(getPresetForStillImageDisplay('Freeze', 'Freeze', 'Freeze', self.rgb(0, 51, 153), self.rgb(255, 0, 0), '18'));
-	presets.push(getPresetForStillImageExit());
+  // Still Images - Display
+  for (let i = 1; i <= 10; i++) {
+    presets[`Display${i}`] = getPresetForStillImageDisplay(
+      `Display Image ${i}`,
+      `Disply ${i}`,
+      `Display${i}`,
+      combineRgb(0, 90, 0),
+      combineRgb(255, 0, 0)
+    );
+  }
+  presets["DisplayTest"] = getPresetForStillImageDisplay(
+    "Display Test Image",
+    "Test image",
+    "DisplayTest",
+    combineRgb(153, 0, 153),
+    combineRgb(255, 0, 0),
+    "18"
+  );
+  presets["Blackout"] = getPresetForStillImageDisplay(
+    "Display Blackout",
+    "Blackout",
+    "Blackout",
+    combineRgb(0, 0, 0),
+    combineRgb(255, 0, 0),
+    "14"
+  );
+  presets["Freeze"] = getPresetForStillImageDisplay(
+    "Freeze",
+    "Freeze",
+    "Freeze",
+    combineRgb(0, 51, 153),
+    combineRgb(255, 0, 0),
+    "18"
+  );
+  presets["ExitImages"] = getPresetForStillImageExit();
+  // Presentation Files
+  presets["PresentationFilesPrev"] = getPresetForPresentationFiles(
+    self.label,
+    "Previous",
+    "prev",
+    combineRgb(0, 0, 0)
+  );
+  presets["PresentationFilesCurr"] = getPresetForPresentationFiles(
+    self.label,
+    "Current",
+    "curr",
+    combineRgb(255, 0, 0)
+  );
+  presets["PresentationFilesNext"] = getPresetForPresentationFiles(
+    self.label,
+    "Next",
+    "next",
+    combineRgb(0, 153, 0)
+  );
 
-    // Presentation Files
-    presets.push(getPresetForPresentationFiles(self.label, 'Previous', 'prev', self.rgb(0, 0, 0)));
-    presets.push(getPresetForPresentationFiles(self.label, 'Current', 'curr', self.rgb(255, 0, 0)));
-    presets.push(getPresetForPresentationFiles(self.label, 'Next', 'next', self.rgb(0, 153, 0)));
+  //Slot Presentations
+  for (let i = 1; i <= 20; i++) {
+    presets[`Slot${i}`] = getPresetforSlotPresentation(
+      self.label,
+      `Slot ${i}`,
+      `slot${i}`,
+      i,
+      combineRgb(0, 0, 0),
+      `Slot${i}`,
+      1,
+      true
+    );
+  }
 
-	//Slot Presentations
-	for (let i = 1; i <= 20; i++) {
-		presets.push(getPresetforSlotPresentation(self.label, `Slot ${i}`, `slot${i}`, i, self.rgb(0, 0, 0), `Slot${i}`, 1, true))
-	}
-
-	return presets;
-}
+  return presets;
+};
 
 function getPresetForPresentationControl(lbl, txt, act, opt = null) {
```

Comment:
I don't request a change here, but you could improve the readability and maintainability of the code if you stick with a preset-like definition object and generate the real preset definitions with the map function instead of many function calls with unannotated parameters.

Reply:
Thank you for your suggestion. but honestly, I am not sure if I got this point clearly + I think it will take some development & testing time, so if possible let's ignore it for now and I will try to improve it later.